<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>632</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <property name="styleSheet">
   <string notr="true"/>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout_2">
    <item row="0" column="0">
     <widget class="QSplitter" name="splitter_2">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <property name="childrenCollapsible">
       <bool>false</bool>
      </property>
      <widget class="QGroupBox" name="grpOperation">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>300</width>
         <height>0</height>
        </size>
       </property>
       <property name="title">
        <string>Operation</string>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <property name="spacing">
         <number>4</number>
        </property>
        <property name="margin">
         <number>4</number>
        </property>
        <item>
         <widget class="QFrame" name="idxSwitcher">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="margin">
            <number>1</number>
           </property>
           <item>
            <widget class="QLabel" name="labOPGrpName">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
               <horstretch>1</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
             <property name="text">
              <string>NVMe Basic Command</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="btnPrev">
             <property name="text">
              <string>&lt;</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="btnNext">
             <property name="text">
              <string>&gt;</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QStackedWidget" name="opStackWidget">
          <property name="frameShape">
           <enum>QFrame::Panel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Raised</enum>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="opNVMeBasic">
           <layout class="QGridLayout" name="gridLayout_4">
            <property name="margin">
             <number>4</number>
            </property>
            <property name="spacing">
             <number>4</number>
            </property>
            <item row="0" column="0">
             <widget class="QTabWidget" name="twBasicCmd">
              <property name="currentIndex">
               <number>1</number>
              </property>
              <widget class="QWidget" name="tabIdentify">
               <attribute name="title">
                <string>Identify</string>
               </attribute>
               <layout class="QVBoxLayout" name="verticalLayout_3">
                <item>
                 <widget class="QFrame" name="frameNSID">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_2">
                   <property name="margin">
                    <number>4</number>
                   </property>
                   <item>
                    <widget class="QLabel" name="label">
                     <property name="text">
                      <string>NSID</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="leNSID">
                     <property name="inputMask">
                      <string>99; </string>
                     </property>
                     <property name="text">
                      <string>1</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QGroupBox" name="grpBoxIdentifySel">
                  <property name="title">
                   <string>CNS</string>
                  </property>
                  <layout class="QVBoxLayout" name="verticalLayout_2">
                   <item>
                    <widget class="QRadioButton" name="rbIdentifyCtrl">
                     <property name="text">
                      <string>Controller</string>
                     </property>
                     <property name="checked">
                      <bool>true</bool>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QRadioButton" name="rbIdentifyNS">
                     <property name="text">
                      <string>Namespace</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="ckbIdentifySave">
                  <property name="text">
                   <string>Save data structure</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnIdentifyExecute">
                  <property name="text">
                   <string>Execute</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="tabGetFeature">
               <attribute name="title">
                <string>Get Feature</string>
               </attribute>
               <layout class="QVBoxLayout" name="verticalLayout_5">
                <item>
                 <widget class="QFrame" name="frameNSID_2">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_3">
                   <property name="margin">
                    <number>4</number>
                   </property>
                   <item>
                    <widget class="QLabel" name="labSEL">
                     <property name="text">
                      <string>SEL</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QComboBox" name="cbSEL">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>1</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QFrame" name="frameNSID_3">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_4">
                   <property name="margin">
                    <number>4</number>
                   </property>
                   <item>
                    <widget class="QLabel" name="labFID">
                     <property name="text">
                      <string>FID</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QComboBox" name="cbFID">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>1</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QGroupBox" name="grpGetLRT">
                  <property name="title">
                   <string>LBA Range Type</string>
                  </property>
                  <layout class="QVBoxLayout" name="verticalLayout_4">
                   <item>
                    <widget class="QCheckBox" name="ckbLRTSave">
                     <property name="text">
                      <string>Save data structure</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QFrame" name="frameNSID_5">
                     <property name="frameShape">
                      <enum>QFrame::NoFrame</enum>
                     </property>
                     <layout class="QHBoxLayout" name="horizontalLayout_6">
                      <property name="margin">
                       <number>4</number>
                      </property>
                      <item>
                       <widget class="QLabel" name="labGetLRTNSID">
                        <property name="text">
                         <string>NSID</string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="leGetLRTNSID">
                        <property name="inputMask">
                         <string>99; </string>
                        </property>
                        <property name="text">
                         <string>1</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </item>
                   <item>
                    <widget class="QFrame" name="frameNSID_4">
                     <property name="frameShape">
                      <enum>QFrame::NoFrame</enum>
                     </property>
                     <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,0">
                      <property name="margin">
                       <number>4</number>
                      </property>
                      <item>
                       <widget class="QLabel" name="labLRTNumber">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
                          <horstretch>1</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="text">
                         <string># of LBA Ranges (0 base) : </string>
                        </property>
                       </widget>
                      </item>
                      <item>
                       <widget class="QLineEdit" name="leGetLRTNum">
                        <property name="sizePolicy">
                         <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
                          <horstretch>0</horstretch>
                          <verstretch>0</verstretch>
                         </sizepolicy>
                        </property>
                        <property name="inputMask">
                         <string>99; </string>
                        </property>
                        <property name="text">
                         <string>0</string>
                        </property>
                       </widget>
                      </item>
                     </layout>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnGetFeatureExecute">
                  <property name="text">
                   <string>Execute</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_2">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </widget>
              <widget class="QWidget" name="tabGetLog">
               <attribute name="title">
                <string>Get Log Page</string>
               </attribute>
               <layout class="QVBoxLayout" name="verticalLayout_6">
                <item>
                 <widget class="QFrame" name="frameNSID_6">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_7">
                   <property name="margin">
                    <number>4</number>
                   </property>
                   <item>
                    <widget class="QLabel" name="labLID">
                     <property name="text">
                      <string>LID</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QComboBox" name="cbLID">
                     <property name="sizePolicy">
                      <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                       <horstretch>1</horstretch>
                       <verstretch>0</verstretch>
                      </sizepolicy>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QFrame" name="frameNSID_7">
                  <property name="frameShape">
                   <enum>QFrame::NoFrame</enum>
                  </property>
                  <layout class="QHBoxLayout" name="horizontalLayout_8">
                   <property name="margin">
                    <number>4</number>
                   </property>
                   <item>
                    <widget class="QLabel" name="labGetLogNSID">
                     <property name="text">
                      <string>NSID</string>
                     </property>
                    </widget>
                   </item>
                   <item>
                    <widget class="QLineEdit" name="leGetLogNSID">
                     <property name="inputMask">
                      <string>99; </string>
                     </property>
                     <property name="text">
                      <string>1</string>
                     </property>
                    </widget>
                   </item>
                  </layout>
                 </widget>
                </item>
                <item>
                 <widget class="QCheckBox" name="ckbLogPageSave">
                  <property name="text">
                   <string>Save data structure</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <widget class="QPushButton" name="btnGetLogPageExecute">
                  <property name="text">
                   <string>Execute</string>
                  </property>
                 </widget>
                </item>
                <item>
                 <spacer name="verticalSpacer_3">
                  <property name="orientation">
                   <enum>Qt::Vertical</enum>
                  </property>
                  <property name="sizeHint" stdset="0">
                   <size>
                    <width>20</width>
                    <height>40</height>
                   </size>
                  </property>
                 </spacer>
                </item>
               </layout>
              </widget>
             </widget>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="opNVMeVendor"/>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QSplitter" name="splitter">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
         <horstretch>1</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="orientation">
        <enum>Qt::Vertical</enum>
       </property>
       <property name="childrenCollapsible">
        <bool>false</bool>
       </property>
       <widget class="QGroupBox" name="grpBoxDevList">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>1</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>NVMe Device List</string>
        </property>
        <layout class="QGridLayout" name="gridLayout">
         <item row="0" column="0">
          <widget class="QTableWidget" name="nvmeDevTable">
           <property name="rowCount">
            <number>0</number>
           </property>
           <property name="columnCount">
            <number>7</number>
           </property>
           <column>
            <property name="text">
             <string>Device Node</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Model Name</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Serial Number</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string># Namespace</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>LBA Size</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Max LBA</string>
            </property>
           </column>
           <column>
            <property name="text">
             <string>Progress</string>
            </property>
           </column>
          </widget>
         </item>
         <item row="1" column="0">
          <widget class="QPushButton" name="btnRefleshDevice">
           <property name="text">
            <string>Enum/ Reflesh</string>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
       <widget class="QGroupBox" name="grpBoxLog">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
          <horstretch>0</horstretch>
          <verstretch>1</verstretch>
         </sizepolicy>
        </property>
        <property name="title">
         <string>Log</string>
        </property>
        <layout class="QGridLayout" name="gridLayout_3">
         <item row="2" column="0">
          <widget class="QPushButton" name="pushButton_3">
           <property name="text">
            <string>PushButton</string>
           </property>
          </widget>
         </item>
         <item row="2" column="1">
          <widget class="QPushButton" name="pushButton_2">
           <property name="text">
            <string>PushButton</string>
           </property>
          </widget>
         </item>
         <item row="2" column="2">
          <widget class="QPushButton" name="btnLogClr">
           <property name="text">
            <string>Clear</string>
           </property>
          </widget>
         </item>
         <item row="0" column="0" colspan="3">
          <widget class="TimeLog" name="log">
           <property name="lineWrapMode">
            <enum>QTextEdit::NoWrap</enum>
           </property>
           <property name="readOnly">
            <bool>true</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>800</width>
     <height>25</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
   </widget>
   <widget class="QMenu" name="menuTools">
    <property name="title">
     <string>Tools</string>
    </property>
    <widget class="QMenu" name="menuDecodeDS">
     <property name="title">
      <string>Decode Data Structure</string>
     </property>
     <addaction name="acDecodeIdentCtrl"/>
     <addaction name="acDecodeIdentNS"/>
     <addaction name="acDecodeLRT"/>
     <addaction name="acDecodeLogErr"/>
     <addaction name="acDecodeLogSmart"/>
     <addaction name="acDecodeLogFWSlot"/>
    </widget>
    <addaction name="menuDecodeDS"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuTools"/>
  </widget>
  <widget class="QStatusBar" name="statusBar"/>
  <action name="acDecodeIdentCtrl">
   <property name="text">
    <string>Identify Controller</string>
   </property>
  </action>
  <action name="acDecodeIdentNS">
   <property name="text">
    <string>Identify Namespace</string>
   </property>
  </action>
  <action name="acDecodeLRT">
   <property name="text">
    <string>LBA Range Type</string>
   </property>
  </action>
  <action name="acDecodeLogErr">
   <property name="text">
    <string>Log Page - Error</string>
   </property>
  </action>
  <action name="acDecodeLogSmart">
   <property name="text">
    <string>Log Page - Smart</string>
   </property>
  </action>
  <action name="acDecodeLogFWSlot">
   <property name="text">
    <string>Log Page - FW Slot</string>
   </property>
  </action>
 </widget>
 <customwidgets>
  <customwidget>
   <class>TimeLog</class>
   <extends>QTextEdit</extends>
   <header>time_log.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
